wp-includes/class-wp-https-ui.php

```diff
@@ -333,4 +334,27 @@ public function upgrade_insecure_requests( $headers ) {
 		$headers['Content-Security-Policy'] = 'upgrade-insecure-requests';
 		return $headers;
 	}
+
+	/**
+	 * Conditionally redirects HTTP requests to HTTPS.
+	 *
+	 * Does not apply if is_admin().
+	 * So if the SSL certificate expires somehow, the admin won't be locked out of wp-admin.
+	 */
+	public function conditionally_redirect_to_https() {
+		$do_redirect = (
+			isset( $_SERVER['REQUEST_SCHEME'], $_SERVER['REQUEST_URI'] ) && 'https' !== $_SERVER['REQUEST_SCHEME']
```

Comment:
@kienstra Is `$_SERVER['REQUEST_SCHEME']` right here? Should it not be using `! is_ssl()`?

Reply:
Good question, @westonruter. I'll get back to you in several hours if that's alright.

Reply:
Hi @westonruter, PR #55 changes this from `$_SERVER['REQUEST_SCHEME']`to `! is_ssl()`. If you have any more suggestions, I could also apply them there.